galaxy_ng/tests/integration/utils/tasks.py

```diff
@@ -10,22 +10,32 @@
     TaskWaitingTimeout
 )
 
+logger = logging.getLogger(__name__)
+
+
+def wait_for_task(api_client, task, timeout=300, raise_on_error=False):
+    if isinstance(task, dict):
```

Comment:
Whats the reason for needing to support two different types for task?

Reply:
there are endpoints that return a task_id (url = f"v3/tasks/{task}/") and others that return a URL that has to be appended to the base url to check the status

Reply:
Could this approach be less confusing than mixing types (while keeping backwards-compatible)?

```python
def wait_for_task(api_client, resp, task_id=None, timeout=300, raise_on_error=False):
    if task_id:
        url = f"v3/tasks/{task_id}/"
    else:
        url = url_safe_join(api_client.config["url"], resp["task"])
```

Then when you have a task_id call it like
```python
wait_for_task(api_client, resp=None, task_id=my_task_id, raise_on_error=True)
```